<launch>
	<arg name="use_pointcloud" default="true"/>
	<arg name="image_topic" />
	<arg name="pointcloud_topic" default=""/>

	<group if="$(arg use_pointcloud)">
		<!-- launch tabletop detection service -->
		<node name="table_object_detection_node" pkg="bwi_perception" type="table_object_detection_node" output="screen">
			<param name="pointcloud_topic" value="$(arg pointcloud_topic)" />
			<param name="up_frame" value="/base_link" />
		</node>


		<!-- launch change detection service -->
		<node name="table_change_detection_node" pkg="bwi_perception" type="table_change_detection_node" output="screen">
			<param name="pointcloud_topic" value="$(arg pointcloud_topic)" />
		</node>

		<!-- launch node for computing features from point clouds -->
		<node name="pointcloud_feature_server" pkg="bwi_perception" type="pointcloud_feature_server" output="screen">
			<param name="pointcloud_topic" value="$(arg pointcloud_topic)" />
		</node>

		<!-- launch node used to set obstacles for arm motion planning -->
		<node name="obstacle_cloud_manager" pkg="bwi_perception" type="obstacle_cloud_manager" output="screen">
			<param name="pointcloud_topic" value="$(arg pointcloud_topic)" />
		</node>
	</group>

</launch>
